keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -334,17 +387,22 @@ In order to ensure backward compatibility for external consumer of the core/v1 E
 |-------------|-------|----------------------------------------------------------------------------|--------------------------------|
 | 1.16        | Alpha | EndpointSliceController (Alpha) EndpointController (GA with normal event)  | Endpoints                      |
 | 1.17        | Beta  | EndpointSliceController (Beta)  EndpointController (GA with warning event) | EndpointSlice                  |
-| 1.18        | GA    | EndpointSliceController (GA)    EndpointController (GA with limitation)    | EndpointSlice                  |
-
-
+| 1.19+       | GA    | EndpointSliceController (GA)    EndpointController (GA with limitation)    | EndpointSlice                  |
 
 
 ## Graduation Criteria
 
-In order to graduate to beta, we need:
+In order to graduate to beta, we will:
 
-- Kube-proxy switch to consume EndpointSlice API. 
-- Verify performance/scalability via testing.
+- Kube-proxy switch to consume EndpointSlice API. (Already done in Alpha)
+- Verify performance/scalability via testing. (Scale tested to 50k endpoints in
+  4k node cluster)
+- Get performance fixes identified in scale testing merged.
+- Implement IPv6 support.
+- Implement e2e tests that ensure both Endpoints and EndpointSlices are tested.
+- Add support for `endpointslice.kubernetes.io/managed-by` label.
+- Add fqdn addressType.
+- Add support for optional appProtocol field on `EndpointPort`.
```

Comment:
need to spec this carefully for extension

Reply:
Should we validate this to only include a set of allowed protocols?

Reply:
I don't see a need. There are an endless number of "application protocols" that we cannot really capture all of them, so we would likely just see users requesting new protocols added to the set constantly, or force to do other hacks to specify protocol as done today.

Reply:
My thinking was that either we specify a known list and everything else is "x-whatever" or we use a label-prefix style name (my company.com/foo) or something.

It's ugly wrt conventions around naming (HTTP all caps vs company.com/name all lowercase).

Don't need to solve it here, do need to solve it I'm the API specs.  One valid answer is "allow anything" (or a specific character set) and don't validate beyond that.

Reply:
My thinking was that either we specify a known list and everything else is "x-whatever" or we use a label-prefix style name (my company.com/foo) or something.

It's ugly wrt conventions around naming (HTTP all caps vs company.com/name all lowercase).

Don't need to solve it here, do need to solve it I'm the API specs.  One valid answer is "allow anything" (or a specific character set) and don't validate beyond that.